seatunnel-server/seatunnel-app/src/main/java/org/apache/seatunnel/app/dal/dao/impl/RoleDaoImpl.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.app.dal.dao.impl;
+
+import org.apache.seatunnel.app.dal.dao.IRoleDao;
+import org.apache.seatunnel.app.dal.entity.Role;
+import org.apache.seatunnel.app.dal.mapper.RoleMapper;
+
+import org.springframework.stereotype.Repository;
+
+import javax.annotation.Resource;
+
+@Repository
+public class RoleDaoImpl implements IRoleDao {
+
+    @Resource
+    private RoleMapper roleMapper;
+
+    @Override
+    public void add(Role role){
+        roleMapper.insert(role);
+    }
+
+    @Override
+    public Role getByRoleName(String roleName) {
+        return roleMapper.selectByRole(roleName);
+    }
+
+}
```

Comment:
Is **Mapper different from **Dao? Are the methods completely duplicated?

Reply:
@CalvinKirs  Do you have any opinion?

Reply:
This is a very common practice in the industry. When we have multiple mapper combinations, it can all be done at the dao layer, and the cache can also be done at the dao layer.

Reply:
Haven't considered using SpringDataJPA? I understand that SpringDataJPA is more suitable for back-end management systems.

Reply:
> This is a very common practice in the industry. When we have multiple mapper combinations, it can all be done at the dao layer, and the cache can also be done at the dao layer.

When we have multiple mapper combinations, we do it more at the service layer, and the combination will involve business logic.

Reply:
Usually the dao layer is only the access interface corresponding to sql, and the business logic will not be involved here.

Reply:
> Haven't considered using SpringDataJPA? I understand that SpringDataJPA is more suitable for back-end management systems.

TBH, personally I prefer to control the SQL myself, I understand we should have some complex queries.

Reply:
> > This is a very common practice in the industry. When we have multiple mapper combinations, it can all be done at the dao layer, and the cache can also be done at the dao layer.
> 
> When we have multiple mapper combinations, we do it more at the service layer, and the combination will involve business logic.

If we want to canche some data, do it at `service` is not suitable, so we need `dao` to be the buffer

Reply:
> 
For the same type of mapper, the dao layer can perform unified transparent transmission. For example, that is, a dao can correspond to multiple mappers, and a combined query can be done at the dao layer instead of using join.

Reply:
> Usually the dao layer is only the access interface corresponding to sql, and the business logic will not be involved here.

I agree

Reply:
> Usually the dao layer is only the access interface corresponding to sql, and the business logic will not be involved here.

For the same type of mapper, the dao layer can perform unified transparent transmission. For example, that is, a dao can correspond to multiple mappers, and a combined query can be done at the dao layer instead of using join.

Reply:
+1

Reply:
For business systems, join is a very bad practice.